src/dataset_processor.py

```diff
@@ -205,10 +205,43 @@ def load_assigned_reads(save_file_name, gffutils_db, multimapped_chr_dict):
         yield current_gene_info, read_storage
 
 
+def construct_models_in_parallel(sample, chr_id, dump_filename, args, multimapped_reads, current_chr_record,
+                                 io_support):
+    processor = DatasetProcessor(args, parallel=True)
+    processor.create_aggregators(sample)
+    gffutils_db = gffutils.FeatureDB(args.genedb, keep_order=True)
+    logger.info("Processing chromosome " + chr_id)
+    transcripts = []
+    tmp_gff_printer = GFFPrinter(sample.out_dir, sample.label, io_support)
+    chr_dump_file = dump_filename + "_" + chr_id
+    for gene_info, assignment_storage in load_assigned_reads(chr_dump_file, gffutils_db, multimapped_reads):
+        for read_assignment in assignment_storage:
+            if read_assignment is None:
+                continue
+            processor.read_stat_counter.add(read_assignment.assignment_type)
+            processor.global_printer.add_read_info(read_assignment)
+            processor.global_counter.add_read_info(read_assignment)
+
+        if not args.no_model_construction:
+            model_constructor = GraphBasedModelConstructor(gene_info, current_chr_record, args, processor.transcript_model_global_counter)
+            model_constructor.process(assignment_storage)
+            tmp_gff_printer.dump(model_constructor)
+            for t in model_constructor.transcript_model_storage:
+                transcripts.append(t.transcript_type)
+
+    logger.info("Finished processing chromosome " + chr_id)
+    processor.global_counter.dump()
+    processor.transcript_model_global_counter.dump()
+    return processor.read_stat_counter, transcripts
+
+
 # Class for processing all samples against gene database
 class DatasetProcessor:
-    def __init__(self, args):
+    def __init__(self, args, parallel=False):
         self.args = args
+        self.read_grouper = create_read_grouper(args)
+        self.io_support = IOSupport(self.args)
+        if parallel: return
```

Comment:
sure - in the assignment step. by the way, it would be nice to use the same names everywhere (now we have read_grouper and read_groupper, for example)